src/test/groovy/com/ofg/uptodate/reporting/NewVersionLoggerSpec.groovy

```diff
@@ -0,0 +1,55 @@
+package com.ofg.uptodate.reporting
+
+import com.ofg.uptodate.LoggerProxy
+import com.ofg.uptodate.dependency.Dependency
+import spock.lang.Specification
+
+import static com.ofg.uptodate.reporting.NewVersionLogger.NEW_VERSIONS_AVAILABLE
+import static com.ofg.uptodate.reporting.NewVersionLogger.NEW_VERSIONS_MESSAGE_HEADER
+import static com.ofg.uptodate.reporting.NewVersionLogger.NO_NEW_VERSIONS_AVAILABLE
+import static com.ofg.uptodate.reporting.NewVersionLogger.NO_NEW_VERSIONS_MESSAGE
+
+class NewVersionLoggerSpec extends Specification {
+
+    static final Dependency HIBERNATE = new Dependency('org.hibernate', 'hibernate-core', '4.3.6.Final')
+    static final Dependency JUNIT = new Dependency('junit', 'junit', '4.11')
+    static final String PROJECT_NAME = 'ProjectName'
+
+    LoggerProxy logger = Mock()
+
+    def 'should report updates in lexical order'() {
+        given:
+            NewVersionLogger reporter = new NewVersionLogger(logger, PROJECT_NAME, false)
+            List<Dependency> updates = [HIBERNATE, JUNIT]
+        when:
+            reporter.reportUpdates(updates as Set)
+        then:
+            1 * logger.lifecycle(_, "$NEW_VERSIONS_MESSAGE_HEADER$JUNIT\n$HIBERNATE")
+    }
+
+    def 'should not print project name by default'() {
+        given:
+            NewVersionLogger reporter = new NewVersionLogger(logger, PROJECT_NAME, false)
+        when:
+            reporter.reportUpdates(dependencies as Set)
+        then:
+            1 * logger.lifecycle(_, message)
+        where:
+            dependencies | message
+            [JUNIT]      | "$NEW_VERSIONS_MESSAGE_HEADER$JUNIT"
+            []           | NO_NEW_VERSIONS_MESSAGE
+    }
+
+    def 'should allow to print project name'() {
+        given:
+            NewVersionLogger reporter = new NewVersionLogger(logger, PROJECT_NAME, true)
+        when:
+            reporter.reportUpdates(dependencies as Set)
+        then:
+            1 * logger.lifecycle(_, message)
+        where:
+            dependencies | message
+            [JUNIT]      | "$NEW_VERSIONS_AVAILABLE for $PROJECT_NAME:\n$JUNIT"
```

Comment:
Maybe it would be better just to type the exact expected message? https://github.com/mockito/mockito/wiki/How-to-write-good-tests#avoid-coding-a-tautology